source/security/mqtt-tls.rst

```diff
@@ -0,0 +1,68 @@
+MQTT-TLS
+==============
+
+This document describes how to configure dojot to use MQTT over TLS.
+
+.. contents:: Table of Contents
+  :local:
+
+tl;dr
+---------
+
+For a device to connect using TLS with Mosquitto, it must posses:
+  - A key pair (.key file)
+  - A certificate signed by a Certificate Authority (CA) trusted by Mosquitto (.crt file)
+  - The certificate of this CA (.crt file)
+  - An entry on Mosquitto Access Control List (ACL), allowing the device to publish on a specific topic.
+  - (optional and not so necessary) A Certificate Revocation List (CRL)
+
+
+When a device is created, Device-manager will (automatic) configure the following components:
+  - IoTAgent -> will registry the new device on its internal cache.
+  - MQTT-Manager -> Will create an entry on the ACL. Allowing the device to publish on a specific topic.
+  - EJBCA -> Will create an end entity so a certificate can be created on the future.
+
+By default, dojot use clear MQTT. To activate TLS, some changes must be done on docker-compose.
+
+On docker-compose.yml
+  - Change the image of the service 'mqtt' from 'ansi/mosquitto' to 'dojot/mqtt-manager'
+  - Change the avalible door for 'mqtt' service from '1883:1883' to '8883:8883'
+  - Change the value of the environment variable MQTT_TLS of the service service 'iotagent' to true (lowercase).
+
+On the configuration file 'iotagent/config.json':
+   - set the flag 'secure' to true
+
+Components
+----------
+
+EJBCA-REST
+***************
+
+EJBCA (https://www.ejbca.org) is a complete Private Key Infrastructure (PKI) capable to manage  CAs, cryptography keys and certificates.
+EJBCA provides a SOAP, web and a command line interface. EJBCA-REST is an wrapper on top of EJBCA that provides modern interfaces, like REST and Kafka.
```

Comment:
perhaps?

EJBCA provides SOAP, web and command line interfaces. EJBCA-REST is a wrapper on top of EJBCA that complements those, allowing the CA to be configured using REST. When used within dojot, it also listens to kafka events, allowing its automatic configuration.

Reply:
Ok